src/matrix.cpp

```diff
@@ -334,29 +332,42 @@ tsym::Vector tsym::Matrix::solveChecked(const Vector& rhs) const
 void tsym::Matrix::compPartialPivots(Vector *b)
 {
     for (size_t j = 0; j + 1 < nCol; ++j) {
-        if (!data[j][j].isZero())
-            continue;
-
-        for (size_t i = j + 1; i < nRow; ++i)
-            if (!data[i][j].isZero()) {
-                swapRows(i, j);
-                if (b != nullptr)
-                    std::swap(b->data[j], b->data[i]);
-                break;
+        size_t highestcomplpos = 0;
+        unsigned highestcompl = data[j][j].getBasePtr()->complexity();
+        for (size_t i = j + 1; i < nRow; ++i){
+            unsigned curr_complexity = data[i][j].getBasePtr()->complexity();
+            if (curr_complexity>highestcompl){
+                highestcompl=curr_complexity;
+                highestcomplpos = i;
             }
+        }
+        if(highestcompl > data[j][j].getBasePtr()->complexity()){
+            swapRows(highestcomplpos, j);
+            if (b != nullptr)
+                std::swap(b->data[j], b->data[highestcomplpos]);
+        }
+        if(data[j][j].isZero())
+            for (size_t i = j + 1; i < nRow; ++i)
+                if (!data[i][j].isZero()){
+                    swapRows(i, j);
+                    if (b != nullptr)
+                        std::swap(b->data[j], b->data[i]);
+                    break;
+                }
     }
 }
 
 void tsym::Matrix::swapRows(size_t index1, size_t index2)
 {
     for (size_t j = 0; j < nCol; ++j)
         std::swap(data[index1][j], data[index2][j]);
+    swapEven = !swapEven;
 }
 
 void tsym::Matrix::factorizeLU()
 {
-    for (size_t j = 0; j + 1 < nCol; ++j) {
-        for (size_t i = j + 1; i < nRow; ++i) {
+    for (size_t j = 0; j + 1 < nCol; ++j){
+        for (size_t i = j + 1; i < nRow; ++i){
```

Comment:
Please be careful to not change existing formatting by accident, could you revert these changes?